chia/full_node/mempool_check_conditions.py

```diff
@@ -44,6 +45,9 @@ def get_flags_for_height_and_constants(height: int, constants: ConsensusConstant
     if height >= constants.SOFT_FORK2_HEIGHT:
         flags = flags | NO_RELATIVE_CONDITIONS_ON_EPHEMERAL
 
+    if height >= constants.SOFT_FORK3_HEIGHT:
+        flags = flags | ENABLE_MESSAGE_CONDITIONS
```

Comment:
I see the thing above but don't we _not_ want this in here?  Shouldn't we just always check message conditions because we know eventually, they will be a failure. Anyone trying to push a spend that will fail then might as well fail now (they likely didn't mean to push it until the soft fork has actually elapsed).

Reply:
> I see the thing above but don't we *not* want this in here? 

We definitely want to enable message conditions after the activation height. That's when we start enforcing them in the consensus rules.

>  Anyone trying to push a spend that will fail then might as well fail now (they likely didn't mean to push it until the soft fork has actually elapsed).

Consider the case where a farmer makes a block that includes an invalid message condition, before the activation height. If some nodes enforce these conditions in *consensus* and some don't, we would split the network.

What you're describing is to enable message conditions in *mempool mode*. That's done by setting this flag as part for `MEMPOOL_MODE`, here: https://github.com/Chia-Network/chia_rs/blob/main/crates/chia-consensus/src/gen/flags.rs#L39

However, it looks like I forgot to set that.

Reply:
I guess I got confused by the name of the file _mempool_ check conditions.  This file affects consensus logic?

Reply:
yes. it was like this when I got here, I promise :)